### ISSUE TITLE
Allow cake bake of controllers to switch between web and API

### ISSUE THREAD

**Issue by liaogz82** [cakephp/cakephp#13077]
This is a (multiple allowed):

* [ ] bug
* [ ] enhancement
* [x] feature-discussion (RFC)

* CakePHP Version: 3.7.
* Platform and Target: LAMP STACK.

### What you did
As usual I run command `bin/cake bake controller [NAME]`.

### What happened
It bakes the usual controller CRUD commands but I am building an API services. I ended up redoing all the functions in the controller but retaining all the scaffolding function names.

### What you expected to happen
It performs as expected, but I was hoping for `bin/cake bake controller api [NAME]` so as to indicate that I am building an API using cake framework, hence doing away the need of Flash component in the controller

**Comment by dereuromark** [comment]
Sounds like you want to use a custom theme then with your API specifics built in.

**Comment by markstory** [comment]
We could include some api focused templates for controllers. That seems like a common use case these days.

**Comment by liaogz82** [comment]
@dereuromark you are mistaken. I want a CRUD scaffolding that is API based, not web based. My app has no use of the view layer as it is talking to a JavaFX client

**Comment by ADmad** [comment]
@liaogz82 He's referring to a custom bake theme which would generate required controller code for APIs, not a theme for your app itself.

**Comment by liaogz82** [comment]
@ADmad noted. I always associate themes to UI since I am working on drupal project now.

**Comment by dereuromark** [comment]
I think we should close this as it can easily be done as plugin solution for the time being and is out of scope for core at this time.
Don't forget to add this to awesome list then for others to also contribute.
